tests/src/Unit/CanadaPostTrackingtTest.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Drupal\Tests\commerce_canadapost\Unit;
+
+/**
+ * Class CanadaPostTrackingtTest.
+ *
+ * @coversDefaultClass \Drupal\commerce_canadapost\Api\TrackingService
+ * @group commerce_canadapost
+ */
+class CanadaPostTrackingtTest extends CanadaPostUnitTestBase {
+
+  /**
+   * ::covers fetchTrackingSummary.
+   */
+  public function testFetchTrackingSummary() {
+    $tracking_summary = $this->trackingService->fetchTrackingSummary('7023210039414604');
+
+    // Test the parsed response.
+    $this->assertArrayHasKey('actual-delivery-date', $tracking_summary);
+    $this->assertArrayHasKey('attempted-date', $tracking_summary);
+    $this->assertArrayHasKey('expected-delivery-date', $tracking_summary);
+    $this->assertArrayHasKey('mailed-on-date', $tracking_summary);
+  }
+
+}
```

Comment:
@shabananavas Here we are not interested in making an actual call to the API and getting a real response. We are not testing the API service, we are testing our own service that processes the response.

What we need to test is whether our service properly parses a successful response or it throws an error as needed in error cases. Look at other tests in the module for examples.